devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp

```diff
@@ -522,6 +514,54 @@ bool WholeBodyDynamicsDevice::openExternalWrenchesPorts(os::Searchable& config)
     return ok;
 }
 
+bool WholeBodyDynamicsDevice::openMultipleAnalogSensorRemapper(os::Searchable &config)
+{
+    // Pass to the remapper just the relevant parameters (sensorList)
+    yarp::os::Property prop;
+    prop.fromString(config.toString().c_str());
+    yarp::os::Property propMASRemapper;
+    yarp::os::Bottle & propMasNames=prop.findGroup("multipleAnalogSensorsNames");
+    propMASRemapper.put("device","multipleanalogsensorsremapper");
+    bool ok=false;
+    for (auto types=1u;types<propMasNames.size();types++){
+        yarp::os::Bottle * mas_type = propMasNames.get(types).asList();
+        if( mas_type->size() != 2 || mas_type->get(1).asList() == nullptr )
+        {
+            yError() << "wholeBodyDynamics: multipleAnalogSensorsNames group is malformed (" << mas_type->toString() << "). ";
+            return false;
+        }
+        else {
+            ok=true;
+        }
+
+        propMASRemapper.addGroup(mas_type->get(0).asString());
+        yarp::os::Bottle  MASnames;
+        yarp::os::Bottle & MASnamesList= MASnames.addList();
+        for(auto i=0u; i < mas_type->get(1).asList()->size(); i++)
+        {
+            MASnamesList.addString(mas_type->get(1).asList()->get(i).asString());
+        }
+        propMASRemapper.put(mas_type->get(0).asString(),MASnames.get(0));
+
+    }
+    ok = multipleAnalogRemappedDevice.open(propMASRemapper);
+    if( !ok )
+    {
+        return ok;
+    }
+    // View relevant interfaces for the multipleAnalogRemappedDevice
+    ok = ok && multipleAnalogRemappedDevice.view(remappedMASInterfaces.temperatureSensors);
+    ok = ok && multipleAnalogRemappedDevice.view(remappedMASInterfaces.ftMultiSensors);
+    ok = ok && multipleAnalogRemappedDevice.view(remappedMASInterfaces.multwrap);
```

Comment:
Could you also explain why we use a IMultipleWrapper interface here? Since, we already have access to the MASInterfaces through the MASRemapper?

Reply:
Not sure I got the question or know the answer since I do not remember much how they work at the moment need to reread some of it

Reply:
I had the same thoughts :D that's why I wanted to understand.

Reply:
In general, remember I copied the same style of controlboardremappers. At the time I was doing it, I understood why now I do not remember hehe